docs/source/search-runs.rst

```diff
@@ -3,207 +3,238 @@
+          )
+
+          mlflow.log_params(
+              {
+                  "batch_size": batch_size[i],
+                  "learning rate": learning_rate[i],
+                  "model": model[i],
+              }
+          )
+
+          mlflow.set_tags(
+              {
+                  "task": task[i],
+                  "environment": environment[i],
+              }
+          )
+
+          dataset = mlflow.data.from_numpy(
+              features=np.random.uniform(size=[20, 28, 28, 3]),
+              targets=np.random.randint(0, 10, size=[20]),
+              name=dataset_name[i],
+              digest=dataset_digest[i],
+          )
+          mlflow.log_input(dataset, context=dataset_context[i])
+
+
+The code above creates 10 MLflow runs with different metrics, params, tags and dataset information. After execution,
+you should find all these runs under the experiment "/search-run-guide", as shown by the screenshot:
+
+.. figure:: ../../../_static/images/search-runs/created-mlflow-runs.png
+   :alt: testing runs
+   :width: 90%
+   :align: center
 
-  .. code-block:: sql
 
-    params.alpha = "0.3" and params.lambda = "0.001" and metrics.error <= 0.05
+.. _search-runs-syntax:
 
+Search Query Syntax
+^^^^^^^^^^^^^^^^^^^
 
-Identifier
-^^^^^^^^^^
+You need to write search queries to fetch your MLflow runs. The query must be a condition or a few conditions
+concatenated by ``and``. At a high level, the syntax is similar to SQL WHERE clause without supporting ``or`` keyword.
+For example, below is a query to search for all runs that have logged a ``loss`` metric smaller than ``1`` to
+MLflow.
 
-Required in the LHS of a search expression. Signifies an entity to compare against.
+.. code-block:: sql
 
-An identifier has two parts separated by a period: the type of the entity and the name of the entity. The type of the entity is ``metrics``, ``params``, ``attributes``, ``datasets``, or ``tags``. The entity name can contain alphanumeric characters and special characters.
+  metric.loss < 1
 
-This section describes supported entity names and how to specify such names in search expressions.
+Briefly a search condition consists of 3 parts:
 
-.. contents:: In this section:
-  :local:
-  :depth: 1
+- An MLflow field on the left side, which can be some metric, param, tag, dataset or MLflow run metadata, e.g.,
+  ``metric.loss`` or ``param.batch_size``.
+- A comparator in the middle, we support 3 types of comparators:
+  - Numeric comparators: ``=``, ``!=``, ``>``, ``>=``, ``<``, and ``<=``.
+  - String comparators: ``=``, ``!=``, ``LIKE`` and ``ILIKE``.
+  - Set comparators: ``IN``.
+- A reference value on the right side.
 
-Entity Names Containing Special Characters
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. note::
+  There is a caveat that using ``IN`` requires wrapping string with single quotes, e.g.,
+  ``params."learning rate" IN ('0.001', '0.01')``. ``params."learning_rate" IN ("0.001", "0.01")`` will throw an error.
```

Comment:
I remember we should use format `params.`learning rate`` ?

Reply:
> that using ``IN`` requires wrapping string with single quotes

Btw why we have this limitation ? I forgot related code details. Is it limitation in `sqlalchemy` parser ?

Reply:
The weird fact is all the following work:
- params.`learning rate`
- params.'learning rate'
- params."learning rate"

I am only mentioning one way in the doc to reduce the users' confusion. 

> Btw why we have this limitation?

I don't know actually... just found this while playing with my code examples and mlflow UI. This is surely a bug to me.